code/go/0chain.net/smartcontract/storagesc/stakepool.go

```diff
@@ -299,7 +299,7 @@ func (sp *stakePool) slash(
 		if err != nil {
 			return 0, err
 		}
-		edbSlash.DelegateRewards[id] = -1 * int64(dpSlash)
+		edbSlash.DelegateRewards[id] = -dpSlash
```

Comment:
Negative value would lead to uint64 overflow of currency.Coin.  I would suggest to leave it as int64 at the moment. Otherwise, we will need to add a new field to indicate whether it's a reward or a penalty, code for updating reward of event db will also need to updated.